esphome/components/esp32_camera/__init__.py

```diff
@@ -124,8 +129,9 @@
 
 
 async def to_code(config):
-    var = cg.new_Pvariable(config[CONF_ID])
-    await setup_entity(var, config)
+    var = cg.new_Pvariable(config[CONF_ID], config[CONF_NAME])
+    cg.add(var.set_disabled_by_default(config[CONF_DISABLED_BY_DEFAULT]))
+    cg.add(var.set_internal(config[CONF_INTERNAL]))
```

Comment:
This can be reverted as handled in `setup_entity`

```suggestion
    var = cg.new_Pvariable(config[CONF_ID])
    await setup_entity(var, config)
```